modules/core/cmake/parallel/detect_openmp.cmake

```diff
@@ -0,0 +1,13 @@
+if(CMAKE_VERSION VERSION_LESS "3.9")
+  message(STATUS "OpenMP detection requires CMake 3.9+")  # OpenMP::OpenMP_CXX target
+endif()
+
+find_package(OpenMP)
```

Comment:
> libraryRelease unload /build/lib/libopencv_core_parallel_openmp.so
> Segmentation fault

`LD_DEBUG=libs` dump is:

```
[  PASSED  ] 1 test.
[ INFO:0] global /home/alalek/projects/opencv/dev/modules/core/src/utils/plugin_loader.impl.hpp (50) libraryRelease unload /home/alalek/projects/opencv/build/opencv/lib/libopencv_core_parallel_openmp.so
    232271:	
    232271:	calling fini: /home/alalek/projects/opencv/build/opencv/lib/libopencv_core_parallel_openmp.so [0]
    232271:	
    232271:	
    232271:	calling fini: /lib64/libgomp.so.1 [0]
    232271:	
Segmentation fault (core dumped)
```

`libgomp.so.1` is de-initialized (`finit` call), but its worker threads are still alive (gdb shows that).

So,
- this is a bug of `libgomp.so.1` (no idea why worker threads are not terminated during unload stage)
- we should not force OpenMP thread pool termination from OpenCV side, because application can use OpenMP directly
- some workaround is required (probably don't unload parallel plugins)

---

<details>

<summary>The similar issue can be observed with TBB</summary>

Issue is sporadic with TBB (not 100% reproducible).

GDB:
```
[ RUN      ] Core_Rand.parallel_for_stable_results
[New Thread 0x7fffd46b7640 (LWP 232758)]
[New Thread 0x7fffa8ffa640 (LWP 232760)]
[New Thread 0x7fffb07f8640 (LWP 232759)]
[       OK ] Core_Rand.parallel_for_stable_results (174 ms)
...
[ INFO:0] global /home/alalek/projects/opencv/dev/modules/core/src/utils/plugin_loader.impl.hpp (50) libraryRelease unload /home/alalek/projects/opencv/build/opencv/lib/libopencv_core_parallel_onetbb.so

Thread 34 "opencv_test_cor" received signal SIGSEGV, Segmentation fault.
(gdb) info threads 
...
  33   Thread 0x7fffd46b7640 (LWP 232758) "opencv_test_cor" 0x00007ffff47ba55d in syscall () from /lib64/libc.so.6
* 34   Thread 0x7fffa8ffa640 (LWP 232760) "opencv_test_cor" 0x00007fffe749ba32 in ?? ()
  35   Thread 0x7fffb07f8640 (LWP 232759) "opencv_test_cor" 0x00007ffff47ba55d in syscall () from /lib64/libc.so.6
```

(`232760` is created during the test run)

LD_DEBUG dump:

```
[ INFO:0] global /home/alalek/projects/opencv/dev/modules/core/src/utils/plugin_loader.impl.hpp (50) libraryRelease unload /home/alalek/projects/opencv/build/opencv/lib/libopencv_core_parallel_onetbb.so
    232863:	
    232863:	calling fini: /home/alalek/projects/opencv/build/opencv/lib/libopencv_core_parallel_onetbb.so [0]
    232863:	
    232863:	
    232863:	calling fini: /lib64/libtbb.so.2 [0]
    232863:	
    232863:	
    232863:	calling fini: /lib64/libirml.so.1 [0]
    232863:	
Segmentation fault (core dumped)
```

</details>

---

TODO: Will try to check some workarounds.

Reply:
Added code to prevent automatic unloading of parallel backend library (including underlying libraries).
There is no huge impact, because plugins are kept loaded until opencv_core unloading anyway.